Dockerfile

```diff
@@ -0,0 +1,134 @@
+COPY --chown=goblin:goblin packages/backend/native-utils/package.json packages/backend/native-utils/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-x64-musl/package.json packages/backend/native-utils/npm/linux-x64-musl/package.json
+COPY --chown=goblin:goblin packages/backend/native-utils/npm/linux-arm64-musl/package.json packages/backend/native-utils/npm/linux-arm64-musl/package.json
+
+# Configure corepack and pnpm, and install dev mode dependencies for compilation
+RUN corepack install -g pnpm@8.15.4
+# && corepack use pnpm@8.12.0
+RUN pnpm i --frozen-lockfile \
+    && pnpm prune
+
+# Copy in the rest of the native-utils rust files
+COPY --chown=goblin:goblin packages/backend/native-utils packages/backend/native-utils/
+
+# Compile native-utils
+RUN pnpm run --filter native-utils build
+
+# Copy in the rest of the files to compile
+COPY --chown=goblin:goblin . ./
+RUN env NODE_ENV=production sh -c "pnpm run --filter '!native-utils' build && pnpm run gulp"
+
+# Trim down the dependencies to only those for production
+RUN pnpm i --prod --frozen-lockfile \
+    && pnpm prune
+
+
+############################# Runtime container
+FROM debian:bookworm-20240211
+ENV PATH="${PATH}:/home/goblin/.npm/bin:/home/goblin/nodejs/bin"
+
+WORKDIR /goblin
+
+# Create a user and install runtime dependencies
+RUN groupadd --gid 1000 goblin \
+    && useradd --uid 1000 --gid goblin --shell /bin/bash --create-home goblin \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        ca-certificates \
+        curl \
+        less \
+        mc \
+        nano \
+        vim \
+        procps \
+        libc6 \
+    && apt-get autoremove -y \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+#RUN apt update && apt install -y --no-install-recommends \
+#    ffmpeg  \
+#    vips-dev  \ #TODO
+#    zip  \
+#    unzip  \
+#    nodejs-current #TODO
+
+COPY . ./
+
+COPY --from=build /home/goblin/nodejs/bin /home/goblin/nodejs/bin
+COPY --from=build /home/goblin/nodejs/lib /home/goblin/nodejs/lib
+COPY --from=build /goblin/packages/megalodon /goblin/packages/megalodon
+
+# Copy node modules
+COPY --from=build /goblin/node_modules /goblin/node_modules
+COPY --from=build /goblin/packages/backend/node_modules /goblin/packages/backend/node_modules
+COPY --from=build /goblin/packages/sw/node_modules /goblin/packages/sw/node_modules
+COPY --from=build /goblin/packages/client/node_modules /goblin/packages/client/node_modules
+COPY --from=build /goblin/packages/firefish-js/node_modules /goblin/packages/firefish-js/node_modules
+
+# Copy the finished compiled files
+COPY --from=build /goblin/built /goblin/built
+COPY --from=build /goblin/packages/backend/built /goblin/packages/backend/built
+COPY --from=build /goblin/packages/backend/assets/instance.css /goblin/packages/backend/assets/instance.css
+COPY --from=build /goblin/packages/backend/native-utils/built /goblin/packages/backend/native-utils/built
```

Comment:
BTW these are the Rust binaries - they're 2 x 5MB, so really puzzled what was the 300MB number @johnHackworth mentioned - either someone cut it down already, or something else was wonky, because I haven't altered the Rust setup. (Besides minor/bugfix package version update.)

Reply:
I think something was seriously wrong with my server when I shared that. It was taking over half an hour to compile the binaries that day, and creating HUGE files... but since then, I haven't seen the issue anymore?  So disregard my previous woes, I have no idea what happened there, but it doesn't seem to happen anymore. Yay computer sciences!